substrafl_examples/go_further/run_diabetes_substrafl.py

```diff
@@ -0,0 +1,561 @@
+# implement only three methods:
+#
+# - ``build_compute_plan(...)``
+# - ``load_local_state(...)``
+# - ``save_local_state(...)``
+#
+# The ``build_compute_plan`` method is essential to create the graph of the compute plan that will be executed on
+# Substra. Using the different ``Nodes`` we created, we will update there states by applying user defined methods.
+#
+# These methods are passed as argument to the ``Node`` using its ``update_state`` method.
+#
+
+
+import numpy as np
+import pandas as pd
+import json
+from collections import defaultdict
+import pandas as pd
+from typing import List, Dict
+
+from substrafl import ComputePlanBuilder
+from substrafl.remote import remote_data, remote
+
+
+class Analytics(ComputePlanBuilder):
+    def __init__(self):
+        super().__init__()
+        self.first_order_aggregated_state = {}
+        self.second_order_aggregated_state = {}
+
+    @remote_data
+    def local_first_order_computation(self, datasamples: pd.DataFrame, shared_state=None):
+        """Compute from the data samples, expected to be pandas dataframe,
+        the means and counts of each column of the data frame.
+        These datasamples or the output of the ``get_data`` function defined
+        in the ``diabetes_substrafl_opener.py`` file are available in the asset
+        folder downloaded at the beginning of the example.
+
+        The signature of a function decorated by @remote_data must contain
+        the datasamples and the shared_state arguments.
+
+        Args:
+            datasamples (pd.DataFrame): Pandas dataframe provided by the opener.
+            shared_state (None, optional): Unused here as this function only
+                use local information already present in the datasamples.
+                Defaults to None.
+
+        Returns:
+            dict: dictionary containing the local information on means, counts
+                and number of sample. This dict will be used as a state to be
+                shared to an AggregationNode in order to compute the aggregation
+                of the different analytics.
+        """
+        df = datasamples
+        states = {
+            "n_samples": len(df),
+            "means": df.select_dtypes(include=np.number).sum().to_dict(),
+            "counts": {
+                name: series.value_counts().to_dict() for name, series in df.select_dtypes(include="category").items()
+            },
+        }
+        return states
+
+    @remote_data
+    def local_second_order_computation(self, datasamples: pd.DataFrame, shared_state: Dict):
+        """This function will use the output of the ``aggregation`` function to compute
+        locally the standard deviation of the different columns.
+
+        Args:
+            datasamples (pd.DataFrame): Pandas dataframe provided by the opener.
+            shared_state (Dict): Output of a first order analytics computation,
+                that must contain the mean.
```

Comment:
```suggestion
                that must contain the means.
```

If I understand correctly, we expect one mean per column